transitions/accept_patient_reports.js

```diff
@@ -13,52 +13,53 @@ const _hasConfig = (doc) => {
     return Boolean(getConfig(doc.form));
 };
 
+// This is more complicated than it needs to be because JS / _ always use ===
+// for equality, complicating the use of unique tuples (i.e. [1,2] !== [1,2])
+const hasGroupTypeCombo = (groupTypeColl, [group, type]) =>
+    groupTypeColl.find(([g, t]) => g === group && t === type);
+
+// This should just be
+//   Object.keys(_.groupBy(tasksToClear, ({group, task}) => [group, task]))
+// but JS only supports strings as keys
+const uniqueGroupTypeCombos = tasks => {
+    const unique = [];
+    tasks.forEach(t => {
+        if (!hasGroupTypeCombo(unique, [t.group, t.type])) {
+            unique.push([t.group, t.type]);
+        }
+    });
+    return unique;
+};
+
 // find the messages to clear
 const findToClear = (registration, reported_date, config) => {
+    // See: https://github.com/medic/medic-docs/blob/master/user/message-states.md#message-states-in-medic-webapp
+    // Both scheduled and pending have not yet been either seen by a gateway or
+    // delivered, so they are both clearable.
+    const typesToClear = ['pending', 'scheduled'];
+
     const reportedDateMoment = moment(reported_date);
     const taskTypes = config.silence_type.split(',').map(type => type.trim());
-    let silence_until;
 
-    if (config.silence_for) {
-        silence_until = reportedDateMoment.clone();
+    const tasksUnderReview = utils.getScheduledTasksByType(registration, taskTypes);
+
+    if (!config.silence_for) {
+        // No range, all clearable tasks should be cleared
+        return tasksUnderReview.filter(task => typesToClear.includes(task.state));
+    } else {
+        // Clear all tasks that are members of a group that "exists" before the
+        // silence_until date. e.g., they have at least one task in their group
+        // whose due date is before silence_until.
+        const silence_until = reportedDateMoment.clone();
```

Comment:
Go camel case for variables.

Reply:
Heh, I didn't even notice. I wonder if it was originally called that because it used to be a config var? Anyway, changed.